ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_op.hpp

```diff
@@ -6,89 +6,128 @@
 
 #include <functional>
 #include <optional>
+#include <variant>
 
+#include "binary_op_type.hpp"
+#include "broadcast_height_and_width_multi_core_program_factory.hpp"
+#include "broadcast_height_multi_core_program_factory.hpp"
+#include "broadcast_width_multi_core_program_factory.hpp"
+#include "element_wise_multi_core_program_factory.hpp"
```

Comment:
I'd like to avoid including them in a header, else we have to rebuild so many things when we change the program